DependencyInjection/HttplugExtension.php

```diff
@@ -359,94 +356,26 @@ private function createUri(ContainerBuilder $container, $serviceId, $uri)
     private function configureAutoDiscoveryClients(ContainerBuilder $container, array $config)
     {
         $httpClient = $config['discovery']['client'];
-
-        if (!empty($httpClient)) {
-            if ($httpClient === 'auto') {
-                $httpClient = $this->registerAutoDiscoverableClient(
-                    $container,
-                    'auto_discovered_client',
-                    $this->configureAutoDiscoveryFactory(
-                        $container,
-                        HttpClientDiscovery::class,
-                        'auto_discovered_client',
-                        $config
-                    ),
-                    $this->isConfigEnabled($container, $config['profiling'])
-                );
-            }
-
-            $httpClient = new Reference($httpClient);
+        if (!empty($httpClient) && $httpClient !== 'auto') {
+            $container->setAlias('httplug.auto_discovery.auto_discovered_client', $httpClient);
+        } elseif (null === $httpClient) {
```

Comment:
Because possible values are `null`, `auto` or any configured client name. So the first case handle the "any configured client name" and the elseif the null case which means remove the definition from the container.